### DISCUSSION TITLE
Organising and displaying timezones

### DISCUSSION THREAD

**Discussion by lolalavender** [code4recovery/12-step-meeting-list#929] [Q&A]
Thank you so much for this plugin. It's really helpful. Are you able to help me with an issue we seem to be having?

Ideally, a new user would click 'meetings' to see the day and time it is now and all of the meetings that are on now, in their own timezone, automatically. Otherwise, they could filter by day and time and choose to see all the meetings in their timezone so they can see all the meetings that are happening now. That's quite far from what we have.

I notice you have that going on beautifully, here: https://aa-intergroup.org/meetings/

For a working example of what we have going on, see here: https://debtorsanonymous.org/meetings/?tsml-day=4 

As you can see, we have several time zones. At the moment, PT is next to ET, next to UK time, etc. 5pm in each of those don't translate to 5pm in all of them.  

To make it user-friendly, how do we consolidate them, so that if someone is looking on the west coast, they (can choose to?) see PT and if it’s east coast, they (can choose to?) see ET. Equally, if they’re looking in Dublin, or Frankfurt, or Iran,  (can choose to?) see their time. Would all be changed to one standard time zone for that to happen? Or is there a way to allow a translation based on location.

At the moment, it's listed in the meeting location timezone, rather than the person who's looking at its timezone.

I've joined a committee to try to help to resolve ux issues. I'm still learning but when I've got more experience, I'm happy to jump on here and offer my service where I can to help.

Any suggestions you have to solve the above are greatly appreciated.

In fellowship,

Lorraine.

**Comment by joshreisner** [comment]
Hi Lorraine! Looks like you've made some pretty extensive customizations to the TSML codebase - is it a fork from a few years ago? I see `3.4.21` as your version number. Looks like you've added new fields, such as `primary_language`, `group_number`, `tz_code`, `tz_description`. Also some fundamental rules have been changed like the requirement to have a geographic location.

The source code for the meeting finder OIAA is using is here: https://github.com/code4recovery/online-meeting-list/ - it is a react/typescript-based system that only supports online meetings and pulls its data indirectly from a Google Sheet. It uses `moment-timezone` to normalize meeting timezones into the user's tz, but they'll be switching to `luxon` soon.

Not sure how to move forward in view of all the effort you've invested in your fork of TSML -- but interested to hear what options you're considering.

**Comment by lolalavender** [comment]
Hey Josh, 

Oh, thank you for that. I wasn't sure what was going on in terms of the customizations, as I'm pretty new. I will pose this at the meeting on Sunday to see what the committee thinks. Do the newer forks have language and a blank space if they wanted to put a group number in? If yes, maybe they'd be happy to switch what we have in favour?

Thank you again for replying. I will let you know where this goes.

Lorraine.

**Reply by joshreisner** [reply]
Not really, no. I think your best bet would be to see if you can get the developer who made those customizations to also sort the meetings by the user's timezone.
